Lib/pdb.py

```diff
@@ -1315,7 +1315,12 @@ def do_commands(self, arg):
         reached.
         """
         if not arg:
-            bnum = len(bdb.Breakpoint.bpbynumber) - 1
+            for bnum in range(len(bdb.Breakpoint.bpbynumber) - 1, -1, -1):
+                if bdb.Breakpoint.bpbynumber[bnum] is not None:
+                    break
+                if bnum == 0:
+                    self.error('no breakpoints set')
+                    return
```

Comment:
@cocolato @aisk Maybe like this?

```suggestion
            for bp in reversed(bdb.Breakpoint.bpbynumber):
                if not bp:
                    continue
                bnum = bp.number
                break
            else:
                self.error('no breakpoints set')
                return
```

Reply:
This looks great! I definitely prefer this approach.